front/src/Phaser/Game/MapEditor/MapEditorModeManager.ts

```diff
@@ -65,21 +65,21 @@ export class MapEditorModeManager {
         this.subscribeToGameMapEvents();
     }
 
-    public executeCommand(type: CommandType, payload: CommandPayload): void {
+    public executeCommand(commandConfig: CommandConfig): void {
         let command: Command;
-        switch (type) {
-            case CommandType.UpdateAreaCommand: {
-                command = new UpdateAreaCommand(this.scene.getGameMap(), payload);
+        switch (commandConfig.type) {
+            case "UpdateAreaCommand": {
+                command = new UpdateAreaCommand(this.scene.getGameMap(), commandConfig);
                 command.execute();
                 // this should not be called with every change. Use some sort of debounce
-                this.scene.connection?.emitMapEditorModifyArea(payload.config);
+                this.scene.connection?.emitMapEditorModifyArea(commandConfig.areaObjectConfig);
                 break;
             }
             default: {
                 return;
             }
```

Comment:
Nice!

Pro tip:

for the default value, you can use:

```
default: {
    const _exhaustiveCheck: never = commandConfig;
}
```

Typescript can never assign a value to a variable typed "never". But because it knows it will never enter this part of the code, the code will run.
If you happen to add a new type to the discriminating union, the compiler will fail to compile until you properly add the "case" statement to handle the new type.

Reply:
Good catch!